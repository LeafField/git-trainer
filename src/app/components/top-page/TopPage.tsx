import React from "react";
import Title from "./top-title/Title";
import Button from "../button/Button";

const TopPage = () => {
  return (
    <div className="flex flex-col justify-center items-center min-h-screen w-full">
      <Title />
      <div className="flex mt-32 md:mt-[12.6rem] justify-center gap-16 md:gap-[9.12rem] ">
        <Button title="Let's Play!" link="/" color="primary" />
        <Button title="How to use" link="/howto" color="secondary" />
      </div>
    </div>
  );
};

export default TopPage;
