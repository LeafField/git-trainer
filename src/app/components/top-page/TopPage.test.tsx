import { render, screen } from "@testing-library/react";
import { composeStories } from "@storybook/react";
import * as story from "./TopPage.stories";

const { TopPageStory } = composeStories(story);

describe("トップページのレンダリングテスト", () => {
  test("top-pageの各Roleの存在確認", () => {
    render(<TopPageStory />);
    expect(
      screen.getByText("Gitコマンドを手軽に空打ちできるサイト"),
    ).toBeInTheDocument();
    expect(
      screen.getByRole("heading", { name: "Git Empty" }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole("link", { name: "Let's Play!" }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole("link", { name: "How to use" }),
    ).toBeInTheDocument();
  });

  test("それぞれのRoleの個数確認", () => {
    render(<TopPageStory />);
    expect(screen.getAllByRole("link")).toHaveLength(2);
    expect(screen.getAllByRole("heading")).toHaveLength(1);
    expect(
      screen.getAllByText("Gitコマンドを手軽に空打ちできるサイト"),
    ).toHaveLength(1);
  });

  test("コピーライト表記がされているか", () => {
    render(<TopPageStory />);
    expect(
      screen.getByText("Copyright © 2023 LeafField All rights reserved."),
    ).toBeInTheDocument();
  });

  test("スナップショットのテスト", () => {
    const { container } = render(<TopPageStory />);
    expect(container).toMatchSnapshot();
  });
});
