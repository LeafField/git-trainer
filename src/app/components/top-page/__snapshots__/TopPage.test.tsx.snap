// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`トップページのレンダリングテスト スナップショットのテスト 1`] = `
<div>
  <div
    class="className"
  >
    <div
      class="flex min-h-screen w-full flex-col items-center justify-center"
    >
      <div
        class="mx-auto flex flex-col gap-5 md:gap-[2.375rem]"
      >
        <p
          class="text-center text-base text-white md:text-2xl "
        >
          Gitコマンドを手軽に空打ちできるサイト
        </p>
        <h1
          class="text-center text-4xl font-bold text-white md:text-8xl "
        >
          Git Empty
        </h1>
      </div>
      <div
        class="mt-32 flex justify-center gap-16 md:mt-[12.6rem] md:gap-[9.12rem] "
      >
        <a
          class="block w-fit rounded-[2.8125rem] px-5 py-2 text-base font-normal text-white after:block after:h-[1px]
      after:w-full after:scale-0 after:transition-all hover:after:scale-100 hover:after:bg-white
      hover:after:content-[""] md:px-10 md:py-4 md:text-2xl
      bg-primary false"
          href="/difficulty"
        >
          Let's Play!
        </a>
        <a
          class="block w-fit rounded-[2.8125rem] px-5 py-2 text-base font-normal text-white after:block after:h-[1px]
      after:w-full after:scale-0 after:transition-all hover:after:scale-100 hover:after:bg-white
      hover:after:content-[""] md:px-10 md:py-4 md:text-2xl
      false border border-white"
          href="/howto"
        >
          How to use
        </a>
      </div>
      <div
        class="absolute bottom-3"
      >
        <footer
          class="pb-10"
        >
          <small
            class="block text-center text-white"
          >
            Copyright © 2023 LeafField All rights reserved.
          </small>
        </footer>
      </div>
    </div>
  </div>
</div>
`;
